File: isni_test.ts
//
// Test our ISNI methods
//
import { assertEquals, assertNotEquals } from "@std/assert";

import { normalizeISNI, validateISNI } from "./isni.ts";
import { verifyISNI } from "./isni_record.ts";

const varified_ids: string[] = [
  "0000 0001 2096 0218",
  "0000 0000 8405 6132",
  "0000 0000 7182 7209",
];

for (const id of varified_ids) {
  // Normalize
  const normalized = normalizeISNI(id);
  assertNotEquals(normalized, undefined);
  console.log(`Normalized ISNI: ${normalized}`);
  // Validate
  assertEquals(validateISNI(id), true);
  // Verify is via isni.org
  const isOK: boolean = await verifyISNI(id);
  assertEquals(isOK, true);
}

File: isni_test.go
package metadatatools

import "testing"

func TestISNI(t *testing.T) {
	if testing.Short() {
		t.Skip("skipping network test in short mode")
	}

	verifiedIDs := []string{
		"0000 0001 2096 0218",
		"0000 0000 8405 6132",
		"0000 0000 7182 7209",
	}

	for _, id := range verifiedIDs {
		normalized := NormalizeISNI(id)
		if normalized == "" {
			t.Errorf("NormalizeISNI(%q) returned an empty string", id)
		}
		t.Logf("Normalized ISNI: %s", normalized)

		if !ValidateISNI(id) {
			t.Errorf("ValidateISNI(%q) = false, expected true", id)
		}

		if !VerifyISNI(id) {
			t.Errorf("VerifyISNI(%q) = false, expected true", id)
		}
	}
}
